### PR TITLE
Additional table filtering options

### LINKED ISSUE #162: More filters for table list API
**Problem**

The table list API should allow filtering. For example, we might want to get the list of all tables in a schema to see if the table the user is trying to create already exists in that schema.

**Proposed solution**

The table list endpoint should support filtering by:
- schema
- before/after: created, last updated
- whether the import was verified

**Additional context**

We should use `django-filter` since it integrates with DRF and makes setting up filters easy.

### PR DESCRIPTION
**Pull request by eito-fis**
Fixes #162 

Add more options for filtering to the table list endpoint.

**Technical details**
Adds an `CharIn` filter on the schema name, a `DateTimeRange` filter on `created_at` and `updated_at`, and a `Boolean` filter on `import_verified`.

The `DateTimeRange` filter allows for several types of filtering:
* An exact time (`?created={time}`)
* A min time (`?created_after={time}`)
* A max time (`?created_before={time}`) 
* A range (`?created_after={start}&created_before={end}`).

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by eito-fis** [comment]
Sidenotes:
* How should merging this PR work? I branched off @mathemancer's branch to use the changes to filtering - should we hold off on merging until the `db_syncing` branch is merged?
* Is there a case where we would want to filter by `import_verified = None`? [Django-filters doesn't support filtering by `None` by default](https://github.com/carltongibson/django-filter/issues/1012), so we would have to make [some changes](https://github.com/carltongibson/django-filter/issues/522#issuecomment-258171253) to support it.

**Comment by mathemancer** [comment]
@eito-fis How difficult would it be to rework this to branch off of `master`?  It will be easier to merge these changes into my branch than the other way, I think (though it's still going to be a pain).  Where is `import_verified` set to `None`?  We should probably just disallow `null` for that, and set a `False` default value.  This will be more logical to me, and would avoid the problem you mentioned.

**Comment by kgodey** [comment]
@mathemancer I had asked @eito-fis to work off of your branch since you'd made changes to the filtering there, my thought was that we could merge this PR in once your DB syncing PR had been merged in to spare you additional merge conflicts. I'm fine with reworking off of `master` if that's easier, just offering more context.

Also, my thought behind `import_verified` being set to `None` was that if a table was created from scratch (rather than from a data file), we'd set `import_verified` to `None` rather than `False`. This way, we can distinguish between tables that don't need to be verified and tables that need to be verified. So yes, we should support filtering by `null`. Also, adding a comment somewhere that that's how we plan to use `import_verified` would be helpful.

**Comment by eito-fis** [comment]
Updated to add an additional `not_imported` parameter. `?not_imported=true` returns all tables with `import_verified = None`, and `?not_imported=false` returns all tables with `import_verified != none`. Having some issues with `django_property_filters` here, and wasn't able to implement this how I wanted. The initial solution of passing `none`, `true`, or `false` to the `import_verified` parameter doesn't seem to work with the library, and the `exclude` option doesn't work with the library (tested), so we have to use `not_imported` instead of just `imported`. Maybe something we can change if `django_property_fitlers` ends up being unneeded.

File: mathesar/tests/views/api/test_table_api.py
Comment (kgodey) [inline mathesar/tests/views/api/test_table_api.py:137]: Since a lot of these assertions are the same, it might be worth pulling them into a separate method so you don't repeat yourself.

**Review by kgodey — APPROVED** [review]
Looks good, added a comment but I don't think it's essential to fix before merge.

I'm not merging since this depends on the `db_syncing` branch being merged first. I'll label as blocked for now.

**Comment by eito-fis** [comment]
Currently blocked by #280

File: mathesar/models.py
Comment (kgodey) [inline mathesar/models.py:31]: Please add a TODO comment that this needs to be removed once we have the proper fix (and also add a note to #280)

**Review by kgodey — APPROVED** [review]

**Pull request merged**
